src/NameApp.java

```diff
@@ -0,0 +1,313 @@
+    }
+
+    //prompts the user for a name and displays the histogram for it
+    private static void displayHistogram(Name[] nameList) {
+        String name = getUsersNameChoice();
+
+        if (searchForName(name, nameList) == -1) {
+            System.out.println("No data for the name " + name);
+        } else {
+            int nameIndex = searchForName(name, nameList);
+            System.out.println(nameList[nameIndex].getHistogram());
+        }
+
+
+    }
+
+    private static void compareTwoNames(Name[] nameList) {
+        String name1 = getUsersNameChoice();
+        String name2 = getUsersNameChoice();
+        int decade = getUsersDecadeChoice();
+        int name1Index = searchForName(name1, nameList);
+        int name2Index = searchForName(name2, nameList);
+        int name1PopRating = nameList[name1Index].getPop(decade);
+        int name2PopRating = nameList[name2Index].getPop(decade);
+        String name1HistoLine = nameList[name1Index].getHistoLine(decade);
+        String name2HistoLine = nameList[name2Index].getHistoLine(decade);
+
+        System.out.println("Data for " + name1);
+        System.out.println("\t" + name1PopRating + ": " + name1HistoLine);
+        System.out.println("Data for " + name2);
+        System.out.println("\t" + name2PopRating + ": "+ name2HistoLine);
+    }
+
+    private static void displayTopTenNames(Name[] nameList) {
+        int decade = getUsersDecadeChoice();
+        int count = 0;
+        int l = 0;
+        String[] decadeName = new String[20];
+        Name[] temp = new Name[nameList.length];
+
+        for (int i = 0; i < nameList.length; i++){
+            temp[i] = nameList[i];
+        }
+
+        for (int index = 0; index < temp.length; index++){
+            int smallestIndex = getIndexOfSmallest(decade, index, temp);
+            Name tempInt = temp[index];
+            temp[index] = temp[smallestIndex];
+            temp[smallestIndex] = tempInt;
+        }
+
+        do {
+
+            if (temp[l].getPop(decade) == 0){
+                l++;
+            }
+            else {
+                decadeName[count] = temp[l].getName() + " " + "(" + temp[l].getPop(decade) + ")";
+
+                count++;
+                l++;
+            }
+
+        } while (count < 20);
+
+        for (int i = 0; i < 20; i += 2){
+            System.out.printf("%20s\t%20s\n", decadeName[i],decadeName[i + 1]);
+        }
+    }
+
+    private static int getIndexOfSmallest(int decade, int startIndex, Name[] temp){
+        int min = temp[startIndex].getPop(decade);
+        int indexOfMin = startIndex;
+
+        for (int index = startIndex + 1; index < temp.length; index++){
+            if (temp[index].getPop(decade) < min){
+                min = temp[index].getPop(decade);
+                indexOfMin = index;
+            }
+        }
+
+        return indexOfMin;
+    }
+
+    //searches for the name the user entered in the array of Name objects
```

Comment:
You might want to look into [Javadoc](https://en.wikipedia.org/wiki/Javadoc), which is pretty much the standard for how to format comments for things like functions and classes.